app/views/layouts/default.phtml

```diff
@@ -1,18 +1,19 @@
 <?php
 
-$protocol       = $this->getParam('protocol', '');
-$domain         = $this->getParam('domain', '');
-$endpoint       = $this->getParam('endpoint', '');
-$platforms      = $this->getParam('platforms', []);
-$version        = $this->getParam('version', '0.0.0');
-$isDev          = $this->getParam('isDev', false);
-$litespeed      = $this->getParam('litespeed', true);
-$analytics      = $this->getParam('analytics', 'UA-26264668-9');
-$mode           = $this->getParam('mode', '');
-$canonical      = $this->getParam('canonical', '');
-$image          = $this->getParam('image', '/images/logo.png');
-$locale         = $this->getParam('locale', null);
-$runtimes       = $this->getParam('runtimes', null);
+$protocol        = $this->getParam('protocol', '');
+$domain          = $this->getParam('domain', '');
+$endpoint        = $this->getParam('endpoint', '');
+$platforms       = $this->getParam('platforms', []);
+$version         = $this->getParam('version', '0.0.0');
+$isDev           = $this->getParam('isDev', false);
+$litespeed       = $this->getParam('litespeed', true);
+$analytics       = $this->getParam('analytics', 'UA-26264668-9');
```

Comment:
Do you happen to know what happens if the tracking ID isn't set? I wonder if it's a better approach to let this be configurable, but default to the Appwrite Core tracking ID. Ideally, if it's blank, there won't be any errors and there won't be any tracking. People who self host appwrite can set the tracking ID to their own tracking ID so they can get analytics themselves.

Reply:
I've looked into blanking the tracking ID and it _kinda_ worked as no [pageview](https://github.com/appwrite/appwrite/blob/dcd731875cf1ddc0835e6a8ddcc421e66cbf00d4/public/scripts/views/analytics/pageview.js#L25) and no [event](https://github.com/appwrite/appwrite/blob/dcd731875cf1ddc0835e6a8ddcc421e66cbf00d4/public/scripts/views/analytics/event.js#L23) analytics got send to Google. Still, two concerns remain with this approach:
1. Google Server will be hit to download the `analytics.js` script even if no tracking ID is provided.
2. [activity analytics](https://github.com/appwrite/appwrite/blob/dcd731875cf1ddc0835e6a8ddcc421e66cbf00d4/public/scripts/views/analytics/activity.js#L19) will be send which while not being send to Google still are analytics.

To resolve the first point, we would need to exclude the GA function in the [`layouts/default.phtml`](https://github.com/appwrite/appwrite/blob/dcd731875cf1ddc0835e6a8ddcc421e66cbf00d4/app/views/layouts/default.phtml#L69) file, which in turn would force us to change the `pageview.js` and `event.js` scripts to handle the missing GA function correctly (these script would currently throw a Reference error in Appwrite's released version if no GA function is in global scope!). To resolve the second point we would need to change the `activity.js` script to prevent sending analytics to appwrite.io. Both of these solutions are kind of what I'm already doing in my implementation.

Nevertheless, I like the idea of providing a custom tracking ID and using the Appwrite ID as the default one and would implement it, if it's wanted.

> Ideally, if it's blank, there won't be any errors...

Currently, the only error thrown is due to the missing GA function. This could easily be resolved by simply not checking for the GA object if analytics is manually disabled.